clients/lsp-terraform.el

```diff
@@ -62,6 +65,30 @@
                   :priority -1
                   :server-id 'tfls))
 
+
+;; terraform-ls
+
+(defgroup lsp-terraform-ls nil
+  "LSP support for Terraform, using terraform-ls from Hashicorp."
+  :group 'lsp-mode
+  :link '(url-link "https://github.com/hashicorp/terraform-ls")
+  :package-version `(lsp-mode . "8.0.1"))
+
+(defcustom lsp-terraform-ls-server "terraform-ls"
+  "Path to the `terraform-ls' binary."
+  :group 'lsp-terraform-ls
+  :risky t
+  :type '(choice
+          (file :tag "File")
+          (repeat string))
+  :package-version `(lsp-mode . "8.0.1"))
+
+(lsp-register-client
+ (make-lsp-client :new-connection (lsp-stdio-connection `(,lsp-terraform-ls-server "serve"))
```

Comment:
we tend to wrap this in lambda to allow changing the property after lsp-mode is loaded. We do so for args.